format/format.go

```diff
@@ -359,10 +381,24 @@ func (w *writer) writeComments(comments []*ast.Comment) {
 }
 
 func (w *writer) writeRules(rules []*ast.Rule, o fmtOpts, comments []*ast.Comment) []*ast.Comment {
+	existingAnnotations, _ := ast.ParseAnnotations(comments)
+
 	for _, rule := range rules {
+
+		if len(existingAnnotations) == 0 {
+			for _, an := range rule.Annotations {
+				w.blankLine()
+				w.writeLine("# METADATA")
+				w.writeComments(an.Comments())
+			}
+		}
+
 		comments = w.insertComments(comments, rule.Location)
```

Comment:
I believe annotation parsing will break if there isn't an empty line separating annotations from trailing comments. `w.insertComments()` ensures there is a blank line between the comments and any annotations written above?

Reply:
Good catch. I guess we can check for trailing comments and add a blank line after writing out the annotations. Updated the code.